React/tic-tac/src/redux/auth/actions.js

```diff
@@ -1,25 +1,22 @@
 import Service from '@services/UserService';
+import { createTypes } from 'redux-recompose';
 
-export const actions = { LOGIN: 'LOGIN', LOGOUT: 'LOGOUT' };
+export const actions = createTypes(['LOGIN', 'LOGOUT'], '@AUTH');
 
-function logIn(payload) {
-  return async dispatch => {
-    const response = await Service(payload);
-    if (response.ok) {
-      dispatch({
-        type: actions.LOGIN,
-        payload: { loggedIn: true }
-      });
-    } else {
-      throw new Error();
-    }
-  };
-}
+const actionCreators = {
+  logIn: payload => ({
+    type: actions.LOGIN,
+    service: Service,
+    payload,
+    target: 'user',
+    successSelector: response => response.data.id,
+    failureSelector: response => true
```

Comment:
What does this function do? 🤔

Reply:
Here the 'success Selector' what it does is indicate the parameter that will be used as a response in the redux